src/components/WelcomeScreen/WelcomeScreen.container.js

```diff
@@ -83,17 +85,40 @@ export class WelcomeScreen extends Component {
             >
               <FormattedMessage {...messages.signUp} />
             </Button>
-            {!isCordova() && (
-              <div className="WelcomeScreen__button WelcomeScreen__button">
-                <GoogleLoginButton
-                  className="WelcomeScreen__button WelcomeScreen__button--google"
-                  onClick={() => {
+
+            <div className="WelcomeScreen__button WelcomeScreen__button">
+              <GoogleLoginButton
+                className="WelcomeScreen__button WelcomeScreen__button--google"
+                onClick={() => {
+                  if (isAndroid()) {
+                    window.plugins.googleplus.login(
+                      {
+                        // 'scopes': '... ', // optional, space-separated list of scopes, If not included or empty, defaults to `profile` and `email`.
+                        webClientId: procces.env.GOOGLE_APP_ID, //'paste the "process.env.GOOGLE_APP_ID" here' envs are not working yet in this repo. On Android, this MUST be included to get an idToken. On iOS, it is not required.
+                        offline: true // optional, but requires the webClientId - if set to true the plugin will also return a serverAuthCode, which can be used to grant offline access to a non-Google server
+                      },
+                      function(obj) {
+                        login(
+                          {
+                            email: 'googletoken',
+                            password: `?access_token=${obj.accessToken}`
+                          },
+                          'oAuth'
+                        );
+                      },
+                      function(msg) {
+                        alert('error: ' + msg);
+                      }
+                    );
+                  } else {
```

Comment:
create a handler function to wrap the code above, we don't use complex functions in the render jsx